include/clock_control.h

```diff
@@ -23,6 +24,20 @@ extern "C" {
 /* Used to select all subsystem of a clock controller */
 #define CLOCK_CONTROL_SUBSYS_ALL	NULL
 
+typedef void (*clock_control_clbk_t)(struct device *dev, void *user_data);
+
+struct clock_control_user {
+	sys_dlist_t list;
```

Comment:
Are you sure you need a dlist here? You could do the same with a slist imo.

Reply:
changed to slist. i didn't think it through as i was preparing RFC just wanted to highlight that structures will be linked together because there can be mutliple requestors that will need to get notified once clock is started.